stub/src/main/java/io/grpc/stub/StreamObserver.java

```diff
@@ -68,6 +68,12 @@
    * exception is thrown by an implementation of {@code onError} no further calls to any method are
    * allowed.
    *
+   * <p>{@code t} should be a {@link io.grpc.StatusException} or {@link
+   * io.grpc.StatusRuntimeException}, but other {@code Throwable} types are possible. Callers should
```

Comment:
As an overrider, I cannot assume Status*Exception because of the exception, so this paragraph isn't helpful. As a caller, I would want to know when I should pass a StatusRuntimeException and when I am qualified for the exception.